[18.0][ADD] account_reconcile_payment_batch_oca

alexis-via opened the pull request:
This module is not really new: it is a "migration" of the module account_reconcile_payment_order developed on Odoo v16 by @etobella  in PR https://github.com/OCA/account-reconcile/pull/722 to the new account.payment.lot object introduced by the OCA module account_payment_batch_oca from  OCA/bank-payment-alternative v18. As a consequence, this module is much smaller than the initial module account_reconcile_payment_order because all the generation of account.payment.lot is made in account_payment_batch_oca.

<img width="1303" height="673" alt="reconcile_payment_lot" src="https://github.com/user-attachments/assets/9da673e6-6d7c-4cd3-b199-0105c532fcfc" />


This PR requires PR https://github.com/OCA/bank-payment-alternative/pull/31 that adds a related field journal_id on account.payment.lot.

etobella committed:
[ADD] account_reconcile_payment_order

etobella committed:
[IMP] account_reconcile_oca: Improve JS for reusing code

alexis-via committed:
[ADD] account_reconcile_payment_batch_oca v18

This new module account_reconcile_payment_batch_oca is a "migration" of
the module account_reconcile_payment_order developped on Odoo v16 by
Enric Tobella (Dixmit) to the new account.payment.lot object introduced
by the OCA module account_payment_batch_oca from
OCA/bank-payment-alternative v18.

etobella reviewed (changes requested):
Add this module at bank payment alternative repo

alexis-via commented:
I proposed this module here because the initial module "account_reconcile_payment_order" was proposed here. But  I'm OK to move this PR to OCA/bank-payment-alternative.
Let's wait one day to see if other people have different opinion about the module's location.

etobella commented:
As specified https://github.com/OCA/account-reconcile/pull/722#issuecomment-2652963083 there is no need for this improvement. With that you just need to add a new view that adds the fields in the filter.

Also, this module was added here originally because there was no "alternative" yet (better times...)

pedrobaeza commented:
No garbage in this repo.

alexis-via commented:
@pedrobaeza I think the word "garbage" is not appropriate. Let's stay nice in this discussion.

I proposed to wait one day for other people to give their opinion on the location of this module. If nobody argues against moving to OCA/bank-payment-alternative, then I'll move the PR and we'll close this one.

pedrobaeza commented:
You didn't play nice, so this shouldn't be here. Go to your "own" repository.

etobella commented:
Also, as it is, this will never be merged. It was stated before that this is not the way. You don't need to modify account_reconcile_oca.

alexis-via commented:
@etobella How do you handle the case where you have a payment order with 3 different execution dates (i.e. 3 different lots) ? In the "Reconcile" tab, if you filter on the payment order name, you will have all the journal items for the 3 execution dates. Could you tell me more about your suggestion to "add a new view that adds the fields in the filter" ? Are you talking about the search view of account.move.line used in account_reconcile_oca ? Are you talking about something else ?

etobella commented:
use the date to filter then, you don't need anything else.

alexis-via commented:
I made a test with an up-to-date account_reconcile_oca. I created a payment order with 2 different execution dates (i.e. 2 lots), with several account.payment for each execution date.

On my first test, there was no "Outstanding payment account" configured on the account.payment.method.line of the bank journal of the payment order. In this case, the solution suggested by @etobella doesn't work: you can't filter on the payment order name in the "Reconcile tab".

For my second test, I added an "Outstanding payment account" on the account.payment.method.line. In the "Reconcile" tab, I added a filter on the payment order name. Then I added a second filter on the date. I didn't find a button to add all the move lines in 1 click... I think that what @etobella refers to when he says "you just need to add a new view that adds the fields in the filter". I suppose it would be a new feature to add to account_reconcile_oca. If we add this feature, I think it should work... but it's not as easy/fast as the solution proposed by this PR:
- with account_reconcile_payment_batch_oca, you can reconcile with 2 clicks (1 click on the "Payment lot" tab and 1 click on the lot)
- with only account_reconcile_oca : 2 clicks + type payment order name and validate + type maturity date and validate

<img width="1301" height="701" alt="pay_rec_native" src="https://github.com/user-attachments/assets/775a656a-ae42-414c-9d1e-ec41678a1b76" />

alexis-via commented:
As suggested by @etobella , this PR has been moved to OCA/bank-payment-alternative: https://github.com/OCA/bank-payment-alternative/pull/32
For some reasons that I cannot explain (but maybe Enric knows why), it works even without the changes in account_reconcile_oca. It seems the JS changes in account_reconcile_oca are not needed in v18...

alexis-via commented:
I forgot to mention an important change: in the new PR, I renamed the module from account_reconcile_payment_batch_oca to account_payment_batch_oca_reconcile

etobella commented:
Renaming is not done this way. Anyway, I don't want to leave my autorship in that module....

alexis-via commented:
@etobella Sorry if I didn't do it "the right way"... I'm not a git expert. I did the move to OCA/bank-payment-alternative that way:

1) I made a "git format-patch" to take your initial commit from OCA/account-reconcile,
2) I made a "git mv" to rename the module in a dedicated commit
3) I made my code changes in the module.

If you want me to do differently, tell me and I'll try my best to do it.

etobella commented:
@alexis-via It is not the first time you say that. I am not a git expert neither, but I try to improve each day.

- I was not a python expert on 2017 (I didn't know how to use Python...) and now I am quite confident with my level (not a top expert, but not so bad)
- I didn't know git on 2017, and now I can handle it.
- I started working in JS in 2020 and now I know how to use it in a lot of places.

I did no courses, just tried many things, read a lot and listened to other OCA contributors. It would be nice if you do that. There is an example of how to do it in the PR where you created the loss of commit history from account_payment_order to your "alternative".

alexis-via commented:
I understand you're referring to the debate in https://github.com/OCA/bank-payment-alternative/pull/5 where you suggested to use a method that re-write commit history. This method has drawbacks (main drawback being changing the reality of the past commits) and Stéphane Bidoul was arguing against this method. I'm also not a fan a re-writing commit history. Even without re-writing commit history, there are still at least 2 methods to keep the full history when migrating to v19:

- use the traditional format-patch method and include the current module name and old module name in the filter
- do like in the mis_builder repo, where the 19.0 branch is pushed from the 18.0 branch.

So I think the method I have used to transfer the code to OCA/bank-payment-alternative is fine.

etobella commented:
It has been used in several places and will be used in edi framework for example. Might have sense to avoid it in repos like mis-builder, but it is necessary in all other modules. Otherwise, commit history will be lost in next migration and that should be even worse in an opensource community.

The way used in mis builder will not be used in all other repos for several reasons, so it is necessary.

pedrobaeza commented:
git always rewrites the commit history (specifically, the commit SHA) in the very moment the parent commit is different, so cherry-picking a commit or rewriting at the same time the folder name is acting the same. If you claim you are not a git expert, why are you arguing about something you don't know?

legalsylvain commented:
<img width="949" height="229" alt="image" src="https://github.com/user-attachments/assets/bdae89b4-367b-4276-9b31-c1681c8a4bbb" />

Hello Pedro,

That's no way to start a conversation. (This is your first message on this PR.)

It's completely aggressive and disrespectful.

There's something you need to work on in terms of how you interact with the community.
We can talk about it together at the next OCA Days, if you like.

Best regards.

pedrobaeza commented:
My message has the source in a lot of time wasted trying to get an agreement and putting a lot from my side with no result. Disrespectful has been all the actions done without agreement, despise all the contributions done in OCA/bank-payment losing them in the fork, etc, etc, so please don't talk without knowing background. I'm very tired of all this crap.

pedrobaeza commented:
@legalsylvain if my message is disrespectful, then what is this: https://github.com/OCA/intrastat-extrastat/pull/258#issuecomment-1935708946? I have been suffering during years this behavior from the same person, not collaborating in anything, doing pings that are totally ignored, not following OCA rules, etc, etc.

legalsylvain commented:
> @legalsylvain if my message is disrespectful, then what is this: https://github.com/OCA/intrastat-extrastat/pull/258#issuecomment-1935708946?

https://github.com/OCA/intrastat-extrastat/pull/258#issuecomment-1935708946 looks also disrespectful.

However, I do not believe that escalating disrespect is a solution here.

> I have been suffering during years this behavior from the same person, not collaborating in anything, doing pings that are totally ignored, not following OCA rules, etc, etc.

This issue should be addressed in a different way. However, I confess that OCA doesn't know how handle such conflict for the time being.

rafaelbn commented:
I would like to kindly remind everyone of the [OCA Code of Conduct](https://odoo-community.org/oca-code-of-conduct), which asks us to keep discussions respectful, constructive, and focused on the technical content.

https://odoo-community.org/oca-code-of-conduct

It is natural that we sometimes disagree on the right approach or on the right repository for a module. However, comments that may be perceived as dismissive or personal can discourage contributors who are genuinely trying to help the community grow.

This PR is not an isolated case: in several recent discussions, similar tensions have arisen, where differences of opinion about process or repository choice escalated into personal remarks. When this happens, the community risks losing valuable contributors and creating an atmosphere of discouragement.

Let’s try to break this pattern by keeping the conversation open, polite, and solution-oriented, so that contributors feel welcome and the technical decisions can be made on their own merits.

Thank you all for your efforts and for the time you dedicate to OCA.

Ping @OCA/board this should be handled at the governance level to ensure consistency and to protect the collaborative spirit of the community.

pedrobaeza commented:
> When this happens, the community risks losing valuable contributors and creating an atmosphere of discouragement.

Yes, me and my team...

rousseldenis commented:
Hi all,

We should calm down the situation as this begins to rise very important frictions in the community.

Thanks to @rafaelbn to remind the code of conduct that should be kept in background of all our interactions as contributors (all of them).

From the @OCA/board point of view:

Apart of technical perspective and historical perspective too, the tone of our writings here and elsewhere within the OCA should stay polite. Extra level discourse should remains behind the keyboard.

In this case:

- @alexis-via hasn't used agressive tone nor @etobella in his answers.
- @pedrobaeza please use polite discourse as everyone else here; moreover, you cannot argue old discussions to justify your tone (as it should justify all your future ones ?) and use threats like:

> When this happens, the community risks losing valuable contributors and creating an atmosphere of discouragement.

> Yes, me and my team...

@pedrobaeza This deserves apologizes to start from a blank page and allows us to go forward.

We all know that there are things to improve on the PSC roles, on governance, etc,  and we know that sometimes frustration kicks in, for us as well. However, we cannot let frustration drive our behavior in life and above all in a community. We all want this community to thrive. We don't gain anything by shouting at each other.

Thanks all for your inputs.

thomaspaulb commented:
If I can also input something here. The gist of the matter is: while technical disagreements can escalate to an emotional or personal level, like they have done here, in the end they should be reduced again to what it is: a technical dispute. So:

- If there are technical reasons to negatively review the PR and not merge it, just state them, or refer back to them if they were already stated
- If someone then ignores them, then ignore the PR

The only reason to then come back to such a PR would be if new, convincing, technical arguments come to the table, and otherwise one can continue politely denying to merge, and if needed, repeating or summarizing the reasons for doing so.

Like @rousseldenis says but in different words perhaps - if we keep on escalating these technical issues to a personal or emotional level it's in the end going to hurt the community. Which is why Code of conduct's are in place - it's not meant to chase away anyone, it's to protect an atmosphere in which technical excellence can thrive. I don't see how anyone (after taking a deep breath) can not agree with that.

The technical dispute itself is another matter.

etobella commented:
Thanks @rousseldenis  and @OCA/board to attend this topic.

I understand that Pedro was harsh at some points in the PR when he closed it, and we might discuss whether an apology is needed. However, this situation is part of a long history, and I believe we should be consistent and ask the same from all actors in similar cases. In the past the board asked me (when I was a board member) to do that in a PR where I was rude (https://github.com/OCA/account-reconcile/pull/658#issuecomment-2129212855). If we ask him to apologize, we need to do the same in a lot of other examples:

- https://github.com/OCA/intrastat-extrastat/pull/258#issuecomment-1935708946 - OCA Stupid Rules
- https://github.com/OCA/bank-payment-alternative/pull/32#issuecomment-3213757494 - Ignoring other PSC comments and rights to decide
- https://github.com/OCA/bank-payment-alternative/pull/5#issuecomment-2991748676 - Ignoring other PSC comments and merging withoug giving time to respond
- https://github.com/OCA/bank-payment-alternative/pull/5#issuecomment-2991911281 - Now I am dramatizing...

I understand that the past is past, but it affects the present and future, so we shouldn't forget it and ammend it if possible.

Regarding the sentence you’re complaining about, I honestly don’t see the issue. He is not threatening anyone; he’s simply expressing that he is tired of this topic and of how he feels about it. I know Pedro is not universally appreciated, but we all know that:

- He is there and answers
- He tries to provide solutions
- You can argue with him and find a common ground (I have done it several times, and I am not working in Tecnativa or for Pedro...)
- He is doing a massive job in the OCA in Porting and Reviewing (Do we need examples of people saying that will not port and Tecnativa is taking care of it?)

That kind of contribution deserves respect. Many others are also doing impressive job too, like @sbidoul with OCA runboat, but Pedro has been doing this for a long time. If there were a “Best Contributor of the Year” award, he would probably deserve it every year (at least since I’ve been involved with the OCA).

So, asking for an apology simply because he expressed his tiredness feels disproportionate to me. That is not a threat, he is exposing his situation, where he feels that the benefits of contributing to the OCA no longer outweigh the difficulties he faces. That is a decision that all Contributors could do at any moment. He is just showing it publicly.

My 2 cts.

simahawk commented:
Dear all, please refer to my recent post on the contributors' ML for the current status after the discussion we had at OCA days.

https://odoo-community.org/groups/contributors-15/contributors-2144870?mode=date&date_begin=&date_end=
